lib/cli/src/commands/create_exe.rs

```diff
@@ -307,9 +306,39 @@ impl CreateExe {
                         .map_err(|err| anyhow::anyhow!(err.to_string()))?;
                     writer.flush()?;
                     drop(writer);
+                    // Write down header file that includes deserialize function
+                    {
+                        let mut writer = BufWriter::new(File::create("static_defs.h")?);
+                        writer.write_all(WASMER_DESERIALIZE_HEADER.as_bytes())?;
+                        writer.flush()?;
+                    }
+
+                    // write C src to disk
+                    let c_src_path = Path::new("wasmer_main.c");
+                    #[cfg(not(windows))]
+                    let c_src_obj = PathBuf::from("wasmer_main.o");
+                    #[cfg(windows)]
+                    let c_src_obj = PathBuf::from("wasmer_main.obj");
 
-                    let cli_given_triple = self.target_triple.clone();
-                    self.compile_c(wasm_object_path, cli_given_triple, output_path)?;
```

Comment:
I think it's better to have this code in a separate function because afaik, the `create-exe-pirita` code also needs to call `compile_c`, so that way we can avoid duplication.

Reply:
compile_c just called LinkCode blah blah. Having it as a different function doesn't let you set flags etc easily, so no.